services/queue/src/dependencytracker.js

```diff
@@ -35,38 +35,46 @@ class DependencyTracker {
    * This will return {message, details} if there is an error.
    */
   async trackDependencies(task) {
-    for (let requiredTaskId of task.dependencies) {
-      await this.db.fns.add_task_dependency(task.taskId, requiredTaskId, task.requires, task.expires);
-    }
+    await this.db.fns.add_task_dependencies(
```

Comment:
this used to be the slowest part - inserting records one by one is slow, especially if you have lots of those

Reply:
Nice! 🎉